### PR TITLE
Align environment variables between consoles

### LINKED ISSUE #29641: Admin Console uses a wrong URL type for auth server
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

The Admin Console seems to be using the admin URL type instead of frontend URL type when configuring auth server URL in the JS adapter.

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

When not logged in, the Admin Console should redirect the user to the login page using frontend URL.

### Actual behavior

When not logged in, the Admin Console redirects the user to the login page using admin URL.

### How to Reproduce?

1. Start the server as:
```
kc.sh start-dev --hostname=http://127.0.0.1.nip.io:8080 --hostname-admin=http://admin.127.0.0.1.nip.io:8080
```
2. Access: http://127.0.0.1.nip.io:8080/admin
3. Browser is redirected to login page at `http://admin.127.0.0.1.nip.io:8080/realms/master/protocol/openid-connect/auth?...`

This behaviour is incorrect. Login flow needs to be always executed at frontend URL, in this case `http://127.0.0.1.nip.io:8080`.

### Anything else?

This caused by https://github.com/keycloak/keycloak/commit/e28aa90fcb6e2603ec1222ca01da18dcde7ede2d.

### PR DESCRIPTION
**Pull request by edewit**
Aligns the `authServerUrl` environment variable between the Admin and Account consoles, so that they both contain the URL to the Keycloak server. This also removes the `authUrl` environment variable (which contains the [URL to the Admin console](https://www.keycloak.org/server/hostname#_administration_console)), since it is not needed anywhere in the Admin console.

Additionally all environment variables are now provided in the same order everywhere, variables shared between consoles first, followed by the specific variables for each console. Redundant variables that are no longer used have been removed.

Closes #29641

### CODE REVIEW

File: js/apps/account-ui/vite.config.ts.timestamp-1717486996414-9949a9e64226d.mjs
Comment (jonkoops) [inline js/apps/account-ui/vite.config.ts.timestamp-1717486996414-9949a9e64226d.mjs:1]: This file should not have been checked in.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/libs/ui-shared/src/context/environment.ts [hunk 1]
```diff
@@ -88,13 +84,32 @@ const defaultEnvironment: AdminEnvironment & AccountEnvironment = {
   },
 };
 
+const defaultAdminEnvironment: AdminEnvironment = {
+  authUrl: "http://localhost:8180",
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply js/libs/ui-shared/src/context/environment.ts]: Resolved in dd1fb27088598b6a430a8e8a533026e32af8926f

File: js/apps/account-ui/vite.config.ts.timestamp-1717486996414-9949a9e64226d.mjs
Comment (jonkoops) [reply js/apps/account-ui/vite.config.ts.timestamp-1717486996414-9949a9e64226d.mjs:1]: Resolved in db742753b2e49c6c30c2861ff20c5b3a90a3e5e5

File: js/libs/ui-shared/src/context/environment.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export type Feature = {
 
 export type BaseEnvironment = {
   /** The URL to the root of the auth server. */
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply js/libs/ui-shared/src/context/environment.ts]: Resolved in a15a2cdd34cf824bf5e681c9b10a0f166012e52d

File: services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java [hunk 3]
```diff
@@ -105,7 +105,7 @@ public Response getMainPage() throws IOException, FreeMarkerException {
 
         URI adminBaseUri = session.getContext().getUri(UrlType.ADMIN).getBaseUri();
         URI authUrl = uriInfo.getBaseUri();
-        map.put("authUrl", authUrl.getPath().endsWith("/") ? authUrl : authUrl + "/");
+        map.put("authServerUrl", authUrl.getPath().endsWith("/") ? authUrl : authUrl + "/");
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply services/src/main/java/org/keycloak/services/resources/account/AccountConsole.java]: Resolved in cf6692c034899982557e463d06c5b60961ddde6f

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc [hunk 4]
```diff
@@ -438,3 +438,22 @@ An external {jdgserver_name} deployment is supported for multi-site setups as ou
...
+* `authUrl`. Use `authServerUrl` instead.
+
+The following variables from the environment script injected into the page of the Account theme are deprecated:
+
+* `authUrl`. Use `authServerUrl` instead.
+* `features.isInternationalizationEnabled`. Do not use this variable.
+
+The following variables were deprecated in the Admin theme:
+
+* `authUrl`. Do not use this variable.
```
Comment (edewit) [inline docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:455]: are you sure about this one? Is it true @vmuzikar, that we only need one url in the admin ui?

Comment (jonkoops) [reply docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:455]: It is not referenced anywhere in the templates or JS code.

**Review by jonkoops — APPROVED** [review]
LGTM. I added a lot of code here myself, so this certainly needs another couple of eyes on it.

**Review by agagancarczyk — APPROVED** [review]
LGTM!

**Review by hmlnarik — APPROVED** [review]
LGTM, thanks!

File: docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc (hunk 4 above)
Comment (edewit) [reply docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc:455]: Yes it was used https://github.com/keycloak/keycloak/pull/30125/files#diff-59cd7df86bacac814b984d485e647e3db6a6011e45a75fbc863bef9d748842fdL27 both `authUrl` and `authServerUrl` are used in the admin ui

**Pull request merged**